components/lib/crash/src/main/java/mozilla/components/lib/crash/service/SentryService.kt

```diff
@@ -57,6 +60,35 @@ class SentryService(
             client.sendMessage(createMessage(crash))
         }
     }
+
+    override fun recordCrashBreadcrumb(crashBreadcrumb: CrashBreadcrumb) {
+        client.context?.recordBreadcrumb(crashBreadcrumb.toSentryBreadcrumb())
+    }
+
+    private fun CrashBreadcrumb.toSentryBreadcrumb(): Breadcrumb {
+        return BreadcrumbBuilder()
+                .setMessage(this.message)
+                .setData(this.data)
+                .setCategory(this.category)
+                .setLevel(this.level.toSentryBreadcrumbLevel())
+                .setType(this.type.toSentryBreadcrumbType())
+                .build()
+    }
+
+    private fun CrashBreadcrumb.Level.toSentryBreadcrumbLevel() = when (this) {
```

Comment:
nit: I'm a test junkie, so I'd make these `internal` so that we can add basic asssertions on the mappings to avoid mistakenly breaking them.

Reply:
How about the visible for test annotation.  Let me know.